src/System.Linq/src/System/Linq/Enumerable.Sum.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace System.Linq
+{
+    public static partial class Enumerable
+    {
+        public static int Sum(this IEnumerable<int> source)
+        {
+            if (source == null) throw Error.ArgumentNull("source");
+            int sum = 0;
+            checked
+            {
+                foreach (int v in source) sum += v;
+            }
+            return sum;
+        }
+
+        public static int? Sum(this IEnumerable<int?> source)
+        {
+            if (source == null) throw Error.ArgumentNull("source");
+            int sum = 0;
+            checked
+            {
+                foreach (int? v in source)
+                {
+                    if (v != null) sum += v.GetValueOrDefault();
```

Comment:
Just use GetValueOrDefault, since we know we are dealing with an int. Add a comment though that we imply an null check here, but not doing it since the result is the same anyways.